docs/src/main/asciidoc/_developer-guide/Embedded_API.adoc

```diff
@@ -0,0 +1,110 @@
+[[embedded-api]]
+= Embedded API
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:idprefix:
+:idseparator:       -
+
+include::../common_attributes.adoc[]
+
+The embedded API can be used to launch {appservername} within a currently running process.
+
+The embedded server can be reinitialized with a different JBoss Home. However the module directory, `module.path`
+system property, and the modules system packages, `jboss.modules.system.pkgs` system property, are effectively static.
+This means that creating a new embedded server or host controller within the same VM will not allow overriding the
+modules directory or the system packages.
+
+By default the logging subsystem is not used to configure logging for the embedded server. A hint can be used on the
+configuration, `Configuration.setEmbeddedLoggingHint(false)`, which allows the logging subsystem to configure logging.
+The system property `java.util.logging.manager` must be set to `org.jboss.logmanager.LogManager` before any attempt is
+made to create a JUL logger.
+
+You can also set a hint to indicate which log manager is being used. The hint attempts to ensure the JBoss Logging
+provider will be compliant with the log manager used.
```

Comment:
Is this only relevant with a particular setting of Configuration.setEmbeddedLoggingHint(...)?

Reply:
Not really no. That said the method `setEmbeddedLoggingHint()` is probably a bad name. It really means "set the `org.wildfly.logging.embedded` system property`. If that value is `true`, the default, then the logging subsystem uses a static `LogContext` for the subsystem to be configured on.

If it's set to `false` then the `LogContext.getLogContext()` is used for the `LogContext`. This was mainly done for offline CLI so the `--stdout` option would work.

Long story short I should probably rename that method to something like `setUseEmbeddedLogContext()`. Not sure I like that name either, but I can see where `setEmbeddedLoggingHint()` is a bit misleading.

That said this specific hint is a bit different. This is used as a short cut to set the `org.jboss.logging.provider` setting and add the module to the system packages. This tells jboss-logging which provider to bind to. It's considered a hint because it can only really be set once.